src/user_crate/mod.rs

```diff
@@ -210,12 +211,76 @@ fn parse_source_and_deps(code_and_deps: &str) -> eyre::Result<(syn::Block, toml:
     let user_dependencies: toml::value::Table =
         toml::from_str(&deps_block).map_err(PlRustError::ParsingDependenciesBlock)?;
 
+    if crate::gucs::allow_listed_dependencies_only() {
+        validate_dependences_are_allowed(&user_dependencies);
+    }
```

Comment:
Per my remarks about separating parsing/validation vs. control: I would actually rather not have an entry point decide whether to run the validation. I would rather attempt to run the validation every time, as even a "no-op" validation step would cause errors if we see a `[dependencies]` table in a format we don't expect and can't handle. That raises the question of whether `allow_listed_dependencies_only` should exist at all.